### ISSUE TITLE
[Feature Request] Allow for steam integrated games (DRM etc)

### ISSUE THREAD

**Issue by fleefie** [dmadisetti/steam-tui#18]
Title says all. It seems that all source engine games are unable to be launched through steam-tui (perhaps due to my install of steamcmd ?)

It seems that the issue is linked to libraries not loading. For Half Life, it's hw.so, for TF2 it's engine.so. Non-source games seem to work just fine.

I've read online that launching these games through the steam client instead of steamcmd worked just fine, perhaps that's the underlying cause, the client pointing the executable to libraries while steam-tui doesn't ?

I can provide more info, screenshots, logs, anything as long as it's requested. Have a good day, reader !

**Comment by dmadisetti** [comment]
So yes, currently steam-tui will not launch games that require some level of DRM or steam library unless steam is running in the background. Currently steam-tui tries to directly launch the binary. This is also not ideal, since there is a steam `run.sh` wrapper that sets environmental variables.

To address other games, we can launch steam without any graphics elements: `steam -console -dev -nofriendsui -no-browser +open "steam://open/xxx"` and then run `app_run <id>` through `steamcmd`

So I see 2 features for this:

- [ ] Use the correct `run.sh` wrapper for games if steam is not running.
- [ ] Allow a user to spawn a headless steam thread (either in options or button press)
   - [ ]  If steam thread is running, then use `app_run <id>`

**Comment by dmadisetti** [comment]
@fleefie I think #21 fixes this. I need to make it a bit more user friendly, but after load, hit spacebar (this'll load steam in the background). Wait a little, and then try launching a game. There'll be no indication that things are working, but that's why I opened #20. Hopefully that should work? I was able to launch a game or two this way.

If you get persistent steam windows launching, take a screenshot (censor your personal info of course)- because that shouldn't be happening.

Thanks!
